app/models/piece.rb

```diff
@@ -1,8 +1,121 @@
-class Piece < ApplicationRecord
-  belongs_to :user
-  belongs_to :game
-
-  def self.types
-    %w[King Queen Rook Bishop Knight Pawn]
-  end
-end
+
+class Piece < ApplicationRecord
+  belongs_to :user
+  belongs_to :game
+
+  def self.types
+    %w[King Queen Rook Bishop Knight Pawn]
+  end
+
+  def obstructed?(destination_x, destination_y)
+    raise 'Error: Invalid Input' if destination_y > 7 || destination_x > 7 || destination_y < 0 || destination_x < 0
+    destination_tests =
+      [
+        horizontal_obstruction_left?(destination_x, destination_y),
+        horizontal_obstruction_right?(destination_x, destination_y),
+        vertical_obstruction_up?(destination_x, destination_y),
+        vertical_obstruction_down?(destination_x, destination_y),
+        diagonal_obstruction_down_left?(destination_x, destination_y),
+        diagonal_obstruction_up_right?(destination_x, destination_y),
+        diagonal_obstruction_down_right?(destination_x, destination_y),
+        diagonal_obstruction_up_left?(destination_x, destination_y)
+      ]
+    raise 'Error: Invalid Input' if !destination_tests.include?(true) && !destination_tests.include?(false)

```

Comment:
We are raising an error here by checking if there is a nil value from our directional tests.  Is there a better way we can write our methods to where they only return true or false, but at the same time keep a test for an invalid move on the board. ie. non-horizontal, vertical or diagonal